src/infer_request.h

```diff
@@ -139,6 +142,7 @@ class InferRequest {
   intptr_t request_address_;
   bool is_decoupled_;
   PreferredMemory preferred_memory_;
+  TRITONSERVER_InferenceTrace* trace_;
```

Comment:
I wonder how this compiles? The infer request class needs to protect the parts that require access to `libtritonserver.so` by `#ifdef TRITON_PB_STUB`. We don't link `libtritonserver.so` to the stub process.

Reply:
It compiles, and I've tested BLS traces with TRITON Tracing APIs and OpenTelemetry, but now after you mentioned that, I'll probably need to verify what happens if `TRITON_PB_STUB` is not defined.